helix-term/src/commands/typed.rs

```diff
@@ -2115,413 +2143,414 @@ pub const TYPABLE_COMMAND_LIST: &[TypableCommand] = &[
             fun: paste_clipboard_after,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
         },
         TypableCommand {
             name: "clipboard-paste-before",
             aliases: &[],
             doc: "Paste system clipboard before selections.",
             fun: paste_clipboard_before,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
         },
         TypableCommand {
             name: "clipboard-paste-replace",
             aliases: &[],
             doc: "Replace selections with content of system clipboard.",
             fun: replace_selections_with_clipboard,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
         },
         TypableCommand {
             name: "primary-clipboard-paste-after",
             aliases: &[],
             doc: "Paste primary clipboard after selections.",
             fun: paste_primary_clipboard_after,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
         },
         TypableCommand {
             name: "primary-clipboard-paste-before",
             aliases: &[],
             doc: "Paste primary clipboard before selections.",
             fun: paste_primary_clipboard_before,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
         },
         TypableCommand {
             name: "primary-clipboard-paste-replace",
             aliases: &[],
             doc: "Replace selections with content of system primary clipboard.",
             fun: replace_selections_with_primary_clipboard,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
         },
         TypableCommand {
             name: "show-clipboard-provider",
             aliases: &[],
             doc: "Show clipboard provider name in status bar.",
             fun: show_clipboard_provider,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
         },
         TypableCommand {
             name: "change-current-directory",
             aliases: &["cd"],
             doc: "Change the current working directory.",
             fun: change_current_directory,
-            completer: Some(completers::directory),
+            completer: CompletionConfig::PositionalArguments(&[Some(completers::directory)]),
         },
         TypableCommand {
             name: "show-directory",
             aliases: &["pwd"],
             doc: "Show the current working directory.",
             fun: show_current_directory,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
         },
         TypableCommand {
             name: "encoding",
             aliases: &[],
             doc: "Set encoding. Based on `https://encoding.spec.whatwg.org`.",
             fun: set_encoding,
-            completer: None,
+            completer: CompletionConfig::NoCompletion,
```

Comment:
Probably not in this  this PR but in the future having an enum completer would be nice for cases like this. An autoamtic way to do that would also be useful for #4411 but I am not sure if serde allows that.

Reply:
Yeah I want to have more metadata about options in general, some discussion in #5939.

Reply:
Yeah that should be left to future work, just something that came to mind while reviewing